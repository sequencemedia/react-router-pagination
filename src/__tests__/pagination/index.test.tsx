import React from 'react'
import renderer from 'react-test-renderer'

import {
  MemoryRouter
} from 'react-router-dom'

import {
  calculateTotalPages,
  calculatePageNumber,
  Pagination,
  Centered,
  Standard
} from 'react-router-pagination/pagination'

describe('react-router-pagination', () => {
  describe('calculateTotalPages()', () => {
    describe('Your data contains 120 items which you want to display at 10 items per page', () => {
      it('returns 12', () => {
        expect(calculateTotalPages(120, 10)).toBe(12)
      })
    })

    describe('Your data contains 60 items which you want to display at 5 items per page', () => {
      it('returns 12', () => {
        expect(calculateTotalPages(60, 5)).toBe(12)
      })
    })

    describe('Your data contains 240 items which you want to display at 20 items per page', () => {
      it('returns 12', () => {
        expect(calculateTotalPages(240, 20)).toBe(12)
      })
    })

    describe('Your data contains 121 items which you want to display at 10 items per page', () => {
      it('returns 13', () => {
        expect(calculateTotalPages(121, 10)).toBe(13)
      })
    })

    describe('Your data contains 61 items which you want to display at 5 items per page', () => {
      it('returns 13', () => {
        expect(calculateTotalPages(61, 5)).toBe(13)
      })
    })

    describe('Your data contains 241 items which you want to display at 20 items per page', () => {
      it('returns 13', () => {
        expect(calculateTotalPages(241, 20)).toBe(13)
      })
    })
  })

  describe('calculatePageNumber()', () => {
    describe('Min', () => {
      it('returns 1', () => {
        expect(calculatePageNumber(0, 12)).toBe(1)
      })
    })

    describe('Max', () => {
      it('returns 12', () => {
        expect(calculatePageNumber(13, 12)).toBe(12)
      })
    })
  })

  describe('<Pagination />', () => {
    describe('With `pageNumber` and `totalPages`', () => {
      it('renders', () => {
        const component = (
          <MemoryRouter>
            <Pagination pageNumber={1} totalPages={2} />
          </MemoryRouter>
        )

        expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })
    })
  })

  describe('<Centered />', () => {
    describe('With `pageNumber` and `totalPages` and `spread`', () => {
      it('renders', () => {
        const component = (
          <MemoryRouter>
            <Centered pageNumber={9} totalPages={9} spread={3} />
          </MemoryRouter>
        )

        expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })
    })
  })

  describe('<Standard />', () => {
    describe('With `pageNumber` and `totalPages` and `spread`', () => {
      it('renders', () => {
        const component = (
          <MemoryRouter>
            <Standard pageNumber={9} totalPages={9} spread={3} />
          </MemoryRouter>
        )

        expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })
    })
  })
})
