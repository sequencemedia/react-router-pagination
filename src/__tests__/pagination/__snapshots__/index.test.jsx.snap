// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-router-pagination <Centered /> With \`pageNumber\` and \`totalPages\` and \`spread\` renders 1`] = `
<ul
  className="pagination"
>
  <li
    className="zero-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/1"
        onClick={[Function]}
        ref={[Function]}
      >
        <span
          className="page-number"
        />
      </a>
    </Link>
  </li>
  <li
    className="reverse-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/6"
        onClick={[Function]}
        ref={[Function]}
      >
        <span
          className="reverse"
        >
          ‹
        </span>
      </a>
    </Link>
  </li>
  <Fragment>
    <li
      className="page"
    >
      <Link>
        <a
          data-discover="true"
          href="/7"
          onClick={[Function]}
          ref={[Function]}
        >
          <span
            className="page-number"
          />
        </a>
      </Link>
    </li>
    <li
      className="page"
    >
      <Link>
        <a
          data-discover="true"
          href="/8"
          onClick={[Function]}
          ref={[Function]}
        >
          <span
            className="page-number"
          />
        </a>
      </Link>
    </li>
    <li
      className="current-page"
    >
      <Link>
        <a
          data-discover="true"
          href="/9"
          onClick={[Function]}
          ref={[Function]}
        >
          <span
            className="page-number"
          />
        </a>
      </Link>
    </li>
  </Fragment>
</ul>
`;

exports[`react-router-pagination <Pagination /> With \`pageNumber\` and \`totalPages\` renders 1`] = `
<ul
  className="pagination"
>
  <Fragment />
  <li
    className="forward-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/1"
        onClick={[Function]}
        ref={[Function]}
      >
        <span
          className="forward"
        >
          ›
        </span>
      </a>
    </Link>
  </li>
  <li
    className="last-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/2"
        onClick={[Function]}
        ref={[Function]}
      >
        <span
          className="page-number"
        />
      </a>
    </Link>
  </li>
</ul>
`;

exports[`react-router-pagination <Standard /> With \`pageNumber\` and \`totalPages\` and \`spread\` renders 1`] = `
<ul
  className="pagination"
>
  <li
    className="zero-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/1"
        onClick={[Function]}
        ref={[Function]}
      >
        <span
          className="page-number"
        />
      </a>
    </Link>
  </li>
  <li
    className="reverse-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/6"
        onClick={[Function]}
        ref={[Function]}
      >
        <span
          className="reverse"
        >
          ‹
        </span>
      </a>
    </Link>
  </li>
  <Fragment>
    <li
      className="page"
    >
      <Link>
        <a
          data-discover="true"
          href="/7"
          onClick={[Function]}
          ref={[Function]}
        >
          <span
            className="page-number"
          />
        </a>
      </Link>
    </li>
    <li
      className="page"
    >
      <Link>
        <a
          data-discover="true"
          href="/8"
          onClick={[Function]}
          ref={[Function]}
        >
          <span
            className="page-number"
          />
        </a>
      </Link>
    </li>
    <li
      className="current-page"
    >
      <Link>
        <a
          data-discover="true"
          href="/9"
          onClick={[Function]}
          ref={[Function]}
        >
          <span
            className="page-number"
          />
        </a>
      </Link>
    </li>
  </Fragment>
</ul>
`;
